lib/client/db/mcp/managedconn.go

```diff
@@ -0,0 +1,164 @@
+package mcp
+
+import (
+	"context"
+	"log/slog"
+	"sync"
+	"time"
+
+	"github.com/gravitational/trace"
+)
+
+// ManagedConn manages a database connection, providing a way to keep a single
+// active connection per database, and also closing the connection once it
+// becomes inactive.
+//
+// Managed connections are always used with Exec function.
+type ManagedConn[T any, C conn[T]] struct {
```

Comment:
Can you document the intent for types `T` and `C`? 

Feels bit weird that the usage below uses `T=pgconn.PgConn` and `C=*pgconn.PgConn`. Perhaps we can have just a single type here?